### PR TITLE
added HeroDevelopmentPage and HeroMenuAsDropdown

### LINKED ISSUE #280: Add HeroDevelopment Page
# ✨ **Hero Development Dashboard**

This feature consolidates all `heroDevelopment` components into a unified, professional, and user-friendly dashboard.

---

### Routing and Localization:
- [ ] Implement routing using **Next.js**
- [ ] Integrate **i18n** to enable multilingual support in all relevant areas.

### Data Visualization:
- [ ] Add a **Pie Chart** to effectively represent data distribution.
- [ ] Incorporate a **Bar Chart** for comparative data visualization.
- [ ] Include a **Pricing Component** to showcase relevant pricing plans or details.
- [ ] Add a **Hero Choosing Dropdown** to allow users to select their desired hero.

### Layout:
- [ ] Develop and refine the **layout** based on the provided **Figma design** to ensure visual consistency and usability.

---

### Unit Testing:
- [ ] Test individual components such as **Pie Chart**, **Bar Chart**, **Pricing Component**, and **Hero Dropdown** for correct rendering and functionality.
- [ ] Validate routing logic using mocked Next.js router to ensure pages render correctly.
- [ ] Write tests to confirm accurate localization behavior when switching between languages.

### Integration Testing:
- [ ] Validate the integration of pie and bar charts with real or mocked data.
- [ ] Ensure the **Pricing Component** and **Hero Dropdown** interact seamlessly with other elements in the dashboard.

### End-to-End Testing:
- [ ] Simulate user interactions, including:
  - Selecting heroes from the dropdown.
  - Interacting with charts and pricing components.
- [ ] Mock API responses for components that rely on external data.
- [ ] Test responsiveness across devices to ensure a consistent user experience.

---

### Resources:
- **Figma Design**: [Altzone Figma Design](https://www.figma.com/design/qaRrNbozqOeESL9lQ2UDrM/Altzone?node-id=0-1&node-type=canvas&t=quMu51uaItriN9lH-0)
- **Testing Tools**: Jest, React Testing Library.

### Notes:
- Ensure the implementation aligns closely with the provided Figma prototype.
- Pay special attention to responsiveness, accessibility, and cross-browser compatibility.
- Mocked API data can be used during development and testing phases if live data is unavailable.

---

### PR DESCRIPTION
**Pull request by juhaj77**
## 📄 **Pull Request Overview**
Added HeroDevelopmentPage and its associated HeroMenuAsDropdown, which differs from the HeroNavMenuAsDropdown component, where the routing logic has been replaced with a callback function. The hero selected from the dropdown is set as the state of HeroDevelopmentPage. The component also has states adjusted by the AttributesPricing component. These states control the content displayed within HeroDevelopmentPage.

closes #280
## 🔧 **Changes Made**

1. [Briefly describe changes you made]
Added initializeHeroGroups hero-specific stat data.
2. [Any refactoring or clean-up tasks]

---

## ✅ **Checklist Before Submission**

- **Functionality**: I have tested my code, and it works as expected.
- **JSDoc**: I have added or updated JSDoc comments for all relevant code.
- **Debugging**: No `console.log()` or other debugging statements are left.
- **Clean Code**: Removed commented-out or unnecessary code.
- **Tests**: Added new tests or updated existing ones for the changes made.
Removed AttributesPricing.test.tsx. The AttributesPricing component has been modified to adjust external states, making it no longer testable as an isolated unit.
- **Documentation**: Documentation has been updated (if applicable).

---

## 📝 **Additional Information**

Provide any additional context or information that reviewers may need to know:

- **Screenshots**: [Include any screenshots or videos if the changes affect the UI]
- **Dependencies**: [Mention any new dependencies or breaking changes]
- **Known Issues**: [List any known issues or limitations]
Warning: Cannot update a component (`HeroDevelopmentPage`) while rendering a different component (`AttributesPricing3`)

heeroKehitys

### CODE REVIEW

File: frontend-next-migration/src/entities/Hero/model/initializeHeroGroups.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import greenBg from '@/shared/assets/images/heros/textBgColors/green2.webp';
 import darkBlue from '@/shared/assets/images/heros/textBgColors/dark-blue_cropped.webp';
 import purpleBg from '@/shared/assets/images/heros/textBgColors/purple.webp';
 import { GroupInfo, HeroGroup, HeroSlug } from '../types/hero';
+import { statData } from '@/entities/Hero/model/stats/statsDataV2';
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Hero/model/initializeHeroGroups.ts:23]: should be "@/entities/Hero" according to https://feature-sliced.design/

File: frontend-next-migration/src/entities/Hero/ui/AttributesPricing3/AttributesPricingV3.tsx [hunk 2]
```diff
@@ -1,11 +1,19 @@
-import { useState, useCallback, useMemo } from 'react';
+import React, { useCallback, useMemo, Dispatch, SetStateAction } from 'react';
 import { Stat } from '../../types/hero';
 import { useClientTranslation } from '@/shared/i18n';
 import { statsPricingData } from '../../model/stats/statsPricingData';
 import { AttributePricingHelper } from '../../model/stats/AttributesPricingHelper';
 import cls from './AttributesPricing.module.scss';
+import { statValue } from '@/entities/Hero/model/stats/statsDataV2';
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Hero/ui/AttributesPricing3/AttributesPricingV3.tsx:7]: should be "@/entities/Hero" according to https://feature-sliced.design/

File: frontend-next-migration/src/features/NavigateHeroes/ui/HeroMenuAsDropdown.test.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { useClientTranslation } from '@/shared/i18n';
+import { HeroSlug } from '@/entities/Hero';
+import HeroMenuAsDropdown from '@/features/NavigateHeroes/ui/HeroMenuAsDropdown';
```
Comment (leolabdev) [inline frontend-next-migration/src/features/NavigateHeroes/ui/HeroMenuAsDropdown.test.tsx:5]: instead "import HeroMenuAsDropdown from '@/features/NavigateHeroes/ui/HeroMenuAsDropdown';"  should be "./HeroMenuAsDropdown"

File: frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx [hunk 4]
```diff
@@ -1,16 +1,189 @@
 'use client';
-import { ReactNode } from 'react';
-import { ComingSoon } from '@/widgets/ComingSoon';
+import React, { useState, useCallback, useMemo } from 'react';
+import Image from 'next/image';
+import cls from './HeroDevelopmentPage.module.scss';
+import { HeroManager, AttributesPie } from '@/entities/Hero';
+import { HeroWithGroup, HeroSlug, Stat } from '@/entities/Hero/types/hero';
+import { BarChart } from '@/entities/Hero/ui/BarChart/BarChart';
```
Comment (leolabdev) [inline frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx:7]: should be "@/entities/Hero" according to https://feature-sliced.design/

File: frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx [hunk 5]
```diff
@@ -1,16 +1,189 @@
 'use client';
-import { ReactNode } from 'react';
-import { ComingSoon } from '@/widgets/ComingSoon';
+import React, { useState, useCallback, useMemo } from 'react';
+import Image from 'next/image';
+import cls from './HeroDevelopmentPage.module.scss';
+import { HeroManager, AttributesPie } from '@/entities/Hero';
+import { HeroWithGroup, HeroSlug, Stat } from '@/entities/Hero/types/hero';
```
Comment (leolabdev) [inline frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx:6]: should be "@/entities/Hero" according to https://feature-sliced.design/

File: frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx [hunk 6]
```diff
@@ -1,16 +1,189 @@
...
+import { BarChart } from '@/entities/Hero/ui/BarChart/BarChart';
+import { color, rarityClassNames } from '@/entities/Hero/model/stats/statsDataV2';
```
Comment (leolabdev) [inline frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx:8]: should be "@/entities/Hero" according to https://feature-sliced.design/

File: frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx [hunk 7]
```diff
@@ -1,16 +1,189 @@
...
+import { statsPricingData } from '@/entities/Hero/model/stats/statsPricingData';
```
Comment (leolabdev) [inline frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx:9]: should be "@/entities/Hero" according to https://feature-sliced.design/

File: frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx [hunk 8]
```diff
@@ -1,16 +1,189 @@
...
+import { useClientTranslation } from '@/shared/i18n';
+import useSizes from '@/shared/lib/hooks/useSizes';
+import { classNames, Mods } from '@/shared/lib/classNames/classNames';
+import { AttributesPricing3 } from '@/entities/Hero/ui/AttributesPricing3/AttributesPricingV3';
```
Comment (leolabdev) [inline frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx:13]: should be "@/entities/Hero" according to https://feature-sliced.design/

File: frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx [hunk 9]
```diff
@@ -1,16 +1,189 @@
...
+import HeroMenuAsDropdown from '@/features/NavigateHeroes/ui/HeroMenuAsDropdown';
```
Comment (leolabdev) [inline frontend-next-migration/src/preparedPages/HeroDevelopmentPage/ui/HeroDevelopmentPage.tsx:14]: should be "@/features/NavigateHeroes" according to https://feature-sliced.design/

**Review by leolabdev — CHANGES_REQUESTED** [review]

**Comment by juhaj77** [comment]
Screenshot from 2024-12-12 21-56-00
 Now it look like this and imports are fixed.

**Pull request merged**